com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Builder.java

```diff
@@ -26,6 +32,28 @@
  */
 public abstract class Builder {
 
+    private static Map<Class<?>,  byte[]> classToParamsDigest = new HashMap<Class<?>,  byte[]>();
+
+    public static void addParamsDigest(Class<?> klass, Map<String, String> options, BuilderParams builderParams) throws NoSuchAlgorithmException {
+        String[] params = builderParams.paramsForSignature();
+        Set<String> paramSet = new HashSet<>(Arrays.asList(params));
+        if (paramSet.add("defoldsdk")) {
```

Comment:
`defoldsdk` sha1 should always be part of the signature for each task. If it's specified as parameter it will be used if not, default one will be used instead see https://github.com/defold/defold/blob/20133ef8a564ae4724ea2ce0fd307be2b0da50a0/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java#L866-L868


Also it's better to do it here to calculate it just once per builder instead of doing it for each task